src/core/algorithms/dd/dd_verifier/dd_verifier.cpp

```diff
@@ -0,0 +1,211 @@
+    }
+    if (type_id == +model::TypeId::kMixed) {
+        throw std::invalid_argument("Column with index \"" + std::to_string(column_index) +
+                                    "\" contains values of different types.");
+    }
+    return column.GetType().IsMetrizable();
+}
+
+double DDVerifier::CalculateDistance(model::ColumnIndex const column_index,
+                                     std::pair<std::size_t, std::size_t> const &tuple_pair) const {
+    double dif = 0;
+    model::TypedColumnData const &column = typed_relation_->GetColumnData(column_index);
+    std::byte const *first_value = column.GetValue(tuple_pair.first);
+    std::byte const *second_value = column.GetValue(tuple_pair.second);
+    auto const &type = static_cast<model::IMetrizableType const &>(column.GetType());
+    dif = type.Dist(first_value, second_value);
+    return dif;
+}
+
+std::vector<std::pair<int, int>> DDVerifier::GetRowsWhereLhsHolds() const {
+    std::vector<std::pair<int, int>> result;
+    std::vector<model::ColumnIndex> columns;
+    for (auto const &constraint : dd_.left) {
+        model::ColumnIndex column_index =
+                typed_relation_->GetSchema()->GetColumn(constraint.column_name)->GetIndex();
+        columns.push_back(column_index);
+    }
+    auto curr_constraint = dd_.left.cbegin();
+    for (std::size_t i = 0; i < num_rows_; i++) {
+        for (std::size_t j = i + 1; j < num_rows_; j++) {
+            if (auto const dif = CalculateDistance(columns[0], {i, j});
+                curr_constraint->constraint.Contains(dif)) {
+                result.emplace_back(i, j);
+            }
+        }
+    }
+    ++curr_constraint;
+    for (std::size_t i = 1; i < columns.size(); i++) {
+        std::vector<std::pair<int, int>> new_result;
+        for (std::size_t j = 0; j < result.size(); j++) {
+            if (double const dif = CalculateDistance(columns[i], result[j]);
+                curr_constraint->constraint.Contains(dif)) {
+                new_result.emplace_back(result[j]);
+            }
+        }
+        result = std::move(new_result);
+        ++curr_constraint;
+    }
+    return result;
+}
+
+void DDVerifier::LoadDataInternal() {
+    typed_relation_ = model::ColumnLayoutTypedRelationData::CreateFrom(*input_table_, false);
+}
+
+void DDVerifier::MakeExecuteOptsAvailable() {
+    using namespace config::names;
+    MakeOptionsAvailable({kDDString});
+}
+
+void DDVerifier::CheckCorrectnessDd() const {
+    for (auto const &constraint : dd_.left) {
+        assert(constraint.constraint.upper_bound >= constraint.constraint.lower_bound);
+        assert(constraint.constraint.lower_bound >= 0);
+        assert(constraint.constraint.upper_bound >= 0);
+    }
+    for (auto const &constraint : dd_.right) {
+        assert(constraint.constraint.upper_bound >= constraint.constraint.lower_bound);
+        assert(constraint.constraint.lower_bound >= 0);
+        assert(constraint.constraint.upper_bound >= 0);
+    }
+}
```

Comment:
Do we have to use assert here?

1. Assertions will be compiled out if `NDEBUG` is set, which is usually the case for release builds.
3. If an assertion fails, it will call `std::abort`, which will abruptly terminate the program. If that's what we want, then go for it, but if not, we might want to consider throwing an exception instead.

Reply:
Smth like:
```cpp
void DDVerifier::CheckCorrectnessDd() const {
    auto check_constraint = [](const auto& constraint) {
        if (constraint.constraint.upper_bound < constraint.constraint.lower_bound) {
            throw std::invalid_argument("Invalid constraint bounds for column: " + 
                                       constraint.column_name);
        }
        if (constraint.constraint.lower_bound < 0 || constraint.constraint.upper_bound < 0) {
            throw std::invalid_argument("Constraint bounds cannot be negative for column: " +
                                       constraint.column_name);
        }
    };

    std::ranges::for_each(dd_.left, check_constraint);
    std::ranges::for_each(dd_.right, check_constraint);
}
```